FinalProject/gs-rest-service-master/complete/src/main/java/services/Imp/UserServiceImp.java

```diff
@@ -0,0 +1,40 @@
+package services.Imp;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import services.UserService;
+import bo.User;
+import dao.UserDao;
+import dao.imp.UserDaoImp;
+
+@Service
+public class UserServiceImp implements UserService {
+
+//	@Autowired
+	UserDao userDao = new UserDaoImp();
+
+	public void addUser(User user) throws SQLException {
+		userDao.addUser(user);
+	}
+
+	public User getUser(String userName) throws SQLException {
+		return userDao.getUser(userName);
+	}
+
+	public void updateUser(User user) throws SQLException {
+		userDao.updateUser(user);
+	}
+
+	public void deleteUser(User user) throws SQLException {

```

Comment:
Delete with REST is by a key, e.g.: with this request: http://www.example.com/users/1234 you're gonna delete user number 1234, so if that is the user id, the method have id param in this method.